src/cmap/connection.ts

```diff
@@ -637,8 +637,17 @@ export function hasSessionSupport(conn: Connection): boolean {
   return description.logicalSessionTimeoutMinutes != null;
 }
 
-function supportsOpMsg(conn: Connection) {
+/** @internal */
+export function supportsOpMsg(conn: Connection, options: { loadBalanced?: boolean }) {
+  // If loadBalanced is true, then we MUST send the initial hello command using OP_MSG.
+  // Since this is the first message and hello/legacy hello hasn't been sent yet,
+  // conn.description will be null and we can't rely on the server check to determine if
+  // the server supports OP_MSG.
   const description = conn.description;
```

Comment:
The stream description is constructed in `Connection.constructor` - so it will never be null (the `description == null` check below is always false).  The stream description has a `loadBalanced` property that is already set and the connection has a `Connection.loadBalanced` property that returns whether the stream description is load balanced. Can we use the description here instead of providing a new option?

Reply:
Per the Slack thread it seems this was the case - the issue was that we were sending `ismaster` over `OP_MSG` instead of `hello`.

Reply:
Ah okay.  So the fix would be in `connect`, where we construct the handshake?

Reply:
It might be that this fix is required:
https://github.com/mongodb/node-mongodb-native/pull/3907/files#diff-7cc25e5d3247913cdf1fe1e7788e951e4435bb1091d6b8aef135c4b171c7c997R217

But the change in supportsOpMsg is not?